model/labels/regexp.go

```diff
@@ -61,9 +64,42 @@ func NewFastRegexMatcher(v string) (*FastRegexMatcher, error) {
 	}
 	m.stringMatcher = stringMatcherFromRegexp(parsed)
 
+	m.matchString = m.compileMatchStringFunction()
 	return m, nil
 }
 
+// compileMatchStringFunction returns the function to run by MatchString().
+func (m *FastRegexMatcher) compileMatchStringFunction() func(string) bool {
+	// If the only optimization available is the string matcher, then we can just run it.
+	if len(m.setMatches) == 0 && m.prefix == "" && m.suffix == "" && m.contains == "" && m.stringMatcher != nil {
+		return m.stringMatcher.Matches
+	}
+
+	return func(s string) bool {
+		if len(m.setMatches) != 0 {
+			for _, match := range m.setMatches {
+				if match == s {
+					return true
+				}
+			}
+			return false
+		}
```

Comment:
We can do this check only once, not per each value checked.

```suggestion
	if len(m.setMatches) != 0 {
		return func(s string) bool {
			for _, match := range m.setMatches {
				if match == s {
					return true
				}
			}
			return false
		}
	}

	return func(s string) bool {
```

Reply:
It's probably one cpu cycle difference, but why not having it when the complexity of the code is the same? :)

Thank you.